src/ng/compile.js

```diff
@@ -2670,7 +2670,9 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
      */
     function addDirective(tDirectives, name, location, maxPriority, ignoreDirective, startAttrName,
                           endAttrName) {
-      if (name === ignoreDirective) return null;
+      if (typeof ignoreDirective === 'string' && name === ignoreDirective) return null;
+      if (typeof ignoreDirective === 'function' && !ignoreDirective.call(null, name)) return null;
```

Comment:
About `ignoreDirective()` returning `true` It was easier for me thinking it like a positive filter returning true if the name was ok.
But since it's called ignoreDirective you're right, we should keep it consistent.